katana/native/cases/views.py

```diff
@@ -0,0 +1,370 @@
+import json
+from utils.navigator_util import Navigator
+from katana_utils import *
+import scanfiles
+
+
+navigator = Navigator();
+path_to_src_python_file = navigator.get_katana_dir() + os.sep + "config.json"
+All_case_action_details = py_file_details(json.loads(open(path_to_src_python_file).read())['pythonsrcdir']);
+setPythonSrcDir(All_case_action_details);
+
+def index(request):
+	path_to_config_file = navigator.get_katana_dir() + os.sep + "config.json"
+	x= json.loads(open(path_to_config_file).read());
+	path_to_testcases = x['xmldir'];
+	template = loader.get_template("listAllCases.html")
+	fpath = path_to_testcases ;
+
+	jtree = navigator.get_dir_tree_json(fpath)
+	jtree['state']= { 'opened': True };
+
+	context = { 
+		'title' 	: 'List of Cases',	
+		'dirpath' : path_to_testcases,
+		'treejs': jtree,
+	}
+	return HttpResponse(template.render(context, request))
+
+def getCaseListTree(request):
+	path_to_config_file = navigator.get_katana_dir() + os.sep + "config.json"
+	x= json.loads(open(path_to_config_file).read());
+	path_to_testcases = x['xmldir'];
+	template = loader.get_template("listAllCases.html")
+	fpath = path_to_testcases ;
+	jtree = navigator.get_dir_tree_json(fpath)
+	jtree['state']= { 'opened': True };
+	return JsonResponse({'treejs': jtree })
+
+##
+## This is very costly imho. 
+##
+## If we keep this information on file on disk then we'll be left behind 
+## if more actions/keywords are added to the dispalys. 
+## On the other hand getting this information on every call is absurd. 
+## However, the spaghetti code in the py_file_details is too hairy to untangle
+## before the demo and is left alone. 
+##
+def getListOfActions(request):
+	"""
+	Returns a list of all the actions and their details using config.json.
+	"""
+	path_to_pythonsrc = json.loads(open(path_to_src_python_file).read())['pythonsrcdir'] ;                 
+	jsr = scanfiles.fetch_action_file_names(path_to_pythonsrc,'driver','all');
+	actions = [ os.path.basename(fn)[:-3] for fn in jsr['ProductDrivers']];
+	actions.insert(0,"To_Be_Developed")
+	return JsonResponse({'actions': actions , 'filesinfo' : All_case_action_details })
+
+
+##
+## If we keep this information on file on disk then we'll be left behind 
+## if more actions/keywords are added to the dispalys. 
+## On the other hand getting this information on every call is absurd. 
+## However, the spaghetti code in the py_file_details is too hairy to untangle
+## before the demo and is left alone. 
+##
+def getListOfKeywords(request):
+	if 0: 
+			path_to_src_python_file = navigator.get_katana_dir() + os.sep + "config.json"
+			x= json.loads(open(path_to_src_python_file).read());
+			path_to_pythonsrc = x['pythonsrcdir']; 
+			details = py_file_details(path_to_pythonsrc);
+	else:
+			details = All_case_action_details;
+
+	responseBack = { 'keywords': ["To_Be_Developed",] }
+	driver = request.GET.get('driver');
+		
+	if driver != "To_Be_Developed":
+			
+		print dir(details);
```

Comment:
Comment out prints if not needed.

Reply:
Agreed. Single line prints only once debug is over.